### ISSUE TITLE
[4.2.dev4] Borderless fullscreen regression vs 4.1.stable

### ISSUE THREAD

**Issue by a-johnston** [godotengine/godot#81553]
### Godot version

v4.2.dev4.mono.official [549fcce5f]

### System information

Godot v4.2.dev4.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3080 Ti (NVIDIA; 31.0.15.3699) - 12th Gen Intel(R) Core(TM) i7-12700K (20 Threads)

### Issue description

In 4.1, it was possible to use non-exclusive borderless fullscreen. However in 4.2.dev4, the same window settings result in a single pixel border. The following screenshot compares the two versions with the same project (the material viewer demo), with the border visible against the skybox:

![borderless_fullscreen_comparison](https://github.com/godotengine/godot/assets/9058941/d2ba86c4-fe36-40f5-9a32-bb4574c98ea6)

Exclusive fullscreen does work without the border in 4.2.dev4 but it comes with the expected flickering when changing windows. Additionally, borderless windowed also works without the border. The issue seems to only affect borderless fullscreen.

### Steps to reproduce

1) Set a project to use non-exclusive fullscreen
2) Toggle "Borderless" to true
3) Run project and observe single pixel border

### Minimal reproduction project

[borderless_issue_example.zip](https://github.com/godotengine/godot/files/12580149/borderless_issue_example.zip)

**Comment by Calinou** [comment]
This is an intentional change performed by https://github.com/godotengine/godot/pull/79016, which was also cherry-picked in 4.1.1.

**Comment by a-johnston** [comment]
Thanks for the quick reply. I did notice a few commits earlier in the 4.x development related to fullscreen optimizations/behaviors but I missed that linked change. I feel like maybe there's some other interaction happening as the issues linked in that PR predate 4.1.stable but borderless fullscreen seemed to work as expected, at least on my system. I hadn't yet downloaded 4.1.1.stable when I made this issue but indeed I am able to reproduce the behavior described in the ticket, so it's definitely unrelated to 4.2 changes (should I update the issue to reflect this?).

I'm not sure if there was an unrelated change landed in 4.1 that fixed the issues separately or if coincidentally there's some other factor involved but I can concretely say on my system 4.1 but not 4.1.1 allows for borderless windowed without flickering etc. It'd be sorta unfortunate if exclusive fullscreen was the only option to avoid the 1px going forward though.
